flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -298,12 +298,40 @@ public Builder cleanupInRocksdbCompactFilter(long queryTimeAfterNumEntries) {
             return this;
         }
 
+        /**
+         * Cleanup expired state while Rocksdb compaction is running.
+         *
+         * <p>RocksDB compaction filter will query current timestamp, used to check expiration, from
+         * Flink every time after processing {@code queryTimeAfterNumEntries} number of state
+         * entries. Updating the timestamp more often can improve cleanup speed but it decreases
+         * compaction performance because it uses JNI call from native code.
+         *
+         * <p>Periodic compaction could speed up expired state entries cleanup, especially for state
+         * entries rarely accessed. Files older than this value will be picked up for compaction,
+         * and re-written to the same level as they were before. It makes sure a file goes through
+         * compaction filters periodically.
+         *
+         * @param queryTimeAfterNumEntries number of state entries to process by compaction filter
+         *     before updating current timestamp
+         * @param periodicCompactionSeconds periodic compaction per seconds which could speed up
+         *     expired state cleanup. 0 means turning off periodic compaction.
+         */
+        @Nonnull
+        public Builder cleanupInRocksdbCompactFilter(
+                long queryTimeAfterNumEntries, long periodicCompactionSeconds) {
```

Comment:
Why the interface is designed with `long` seconds?
From my point of view, a `Duration` parameter is better for human read.
Current value of `0xfffffffffffffffeL` looks really strange.

Reply:
Sorry for the delayed update due to my personal business last month.
Thanks for the suggestion. I replaced `long` with `Time` which is also the type of TTL.